numpy/testing/nosetester.py

```diff
@@ -411,7 +411,20 @@ def test(self, label='fast', verbose=1, extra_argv=None,
                 warnings.simplefilter("always")
                 from ..distutils import cpuinfo
             sup.filter(category=UserWarning, module=cpuinfo)
-
+            # See #7949: Filter out deprecation warnings due to the -3 flag to
+            # python 2
+            if sys.version_info.major == 2 and sys.py3kwarning:
+                # This is very specific, so using the fragile module filter
+                # is fine
+                import threading
+                sup.filter(DeprecationWarning,
+                           r"sys\.exc_clear\(\) not supported in 3\.x",
+                           module=threading)
+                sup.filter(DeprecationWarning, message="in 3\.x, __setslice__")
+                sup.filter(DeprecationWarning, message="in 3\.x, __getslice__")
+                sup.filter(DeprecationWarning, message="buffer\(\) not supported in 3\.x")
+                sup.filter(DeprecationWarning, message="CObject type is not supported in 3\.x")
+                sup.filter(DeprecationWarning, message="comparing unequal types not supported in 3\.x")
```

Comment:
I might still have a look what the last two things filter later, but I trust you its all good.